e_commerce_app/ItemDetailView/ItemDetailViewController.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  ItemDetailViewController.swift
+//  e_commerce_app
+//
+//  Created by Bhavin Kapadia on 2022-10-16.
+//
+
+import Foundation
+import UIKit
+
+class ItemDetailViewController: UIViewController {
+    @IBOutlet weak var itemDetailViewItemPrice: UILabel!
+    @IBOutlet weak var itemDetailViewImage: UIImageView!
+    @IBOutlet weak var itemDetailViewTitle: UILabel!
+    @IBOutlet weak var itemDetailViewDescription: PaddingLabel!
+    
+    private let itemDetailViewModel: ItemDetailViewModel
+    
+    
+    init?(itemDetailViewModel: ItemDetailViewModel, coder: NSCoder) {
+        self.itemDetailViewModel = itemDetailViewModel
+        super.init(coder: coder)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    lazy var containerView: UIView = {
+        let view = UIView()
+        view.backgroundColor = .lightGray
+        return view
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let itemPrice = itemDetailViewModel.product.price ?? 0.00 as Double
+        let doubleItemPriceString = String(format: "%.2f", itemPrice)
+        
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "Cancel", style: .plain, target: nil, action: nil)
+        self.itemDetailViewDescription.text = itemDetailViewModel.product.productDescription ?? ""
+        self.itemDetailViewTitle.text = itemDetailViewModel.product.title ?? ""
+        self.itemDetailViewItemPrice.text = "$ \(doubleItemPriceString)"
+        self.navigationController?.navigationBar.topItem?.backButtonTitle = "Return Home"
+        
+        if let productImageURL = itemDetailViewModel.product.image{
```

Comment:
This logic could be moved to the VM. All these VM refactors can be done when you're at the end of the project and everything is working.